sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -240,6 +240,15 @@ object SQLConf {
     .intConf
     .createWithDefault(100)
 
+  val MULTI_COMMUTATIVE_OP_OPT_THRESHOLD =
+    buildConf("spark.sql.analyzer.canonicalization.multiCommutativeOpMemoryOptThreshold")
+      .internal()
+      .doc("The minimum number of consecutive commutative expressions to" +
+        " invoke the MultiCommutativeOp memory optimization during canonicalization.")
+      .version("3.3.0")
```

Comment:
This version should be `3.5.0` or `3.4.0` depending on where we want to merge this PR.